WalletWasabi/KeyManagement/BlockchainState.cs

```diff
@@ -0,0 +1,34 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using WalletWasabi.JsonConverters;
+using WalletWasabi.Models;
+
+namespace WalletWasabi.KeyManagement
+{
+	[JsonObject(MemberSerialization.OptIn)]
+	public class BlockchainState
+	{
+		[JsonProperty(Order = 0)]
+		[JsonConverter(typeof(HeightJsonConverter))]
+		public Height BestHeight { get; set; }
```

Comment:
```suggestion
				public Height BestHeight => BlockStates.Last()?.BlockHeight ?? 0; 
```

Reply:
The best height is not the same as the blockstate's latest block's height. With your change we'd be constantly removing too many blockstates at serialization: https://github.com/zkSNACKs/WalletWasabi/pull/1331/files#diff-76400d6b99f6bb5082f33bdd8144c9d0R182